src/VisualStudio/Core/Impl/Options/OptionBinding.cs

```diff
@@ -1,34 +1,41 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.ComponentModel;
 using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.Options
 {
-    internal class OptionBinding<T>
+    internal class OptionBinding<T> : INotifyPropertyChanged
     {
-        private readonly IOptionService _optionService;
+        private readonly OptionStore _optionStore;
         private readonly Option<T> _key;
 
-        public OptionBinding(IOptionService optionService, Option<T> key)
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public OptionBinding(OptionStore optionStore, Option<T> key)
         {
-            _optionService = optionService;
+            _optionStore = optionStore;
             _key = key;
+
+            _optionStore.OptionChanged += (sender, e) =>
```

Comment:
Does this ever get unsubscribed? We don't re-create these if I dismiss and reopen the pages, correct?

Reply:
The pages that create the bindings are create once and live for the lifetime of the application.